k8s/postgresql/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,141 @@
 actions:
-- name: dummy test
+- name: secure login
   bashTest:
     script: |-
-      exit 0
+      SERVICE="${APP_INSTANCE_NAME}-postgresql-svc"
+      IP=''
+      PGPASSWORD=''
+      FAILED=0
+
+      function wait_for_service_external_ip {
+        IP=$(kubectl get svc "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.status.loadBalancer.ingress[0].ip}")
+        if [[ $? -ne 0 ]]; then
+          IP=""
+        fi
+      }
+
+      function get_postgresql_postgres_password {
+        PGPASSWORD=$(kubectl get secret "${APP_INSTANCE_NAME}-secret" \
+          --namespace "${NAMESPACE}" \
+          --output=jsonpath='{.data.password}' | base64 -d)
+        if [[ $? -ne 0 ]]; then
+          PGPASSWORD=""
+        fi
+      }
+
+      echo "Waiting for Service external ip address."
+      wait_for_service_external_ip
+      while [[ "${IP}" == "" ]]; do
+        echo "Not yet IP"
+        sleep 20
+        wait_for_service_external_ip
+      done
+
+      echo "Waiting for PostgreSQL postgres Password."
+      get_postgresql_postgres_password
+      while [[ "${PGPASSWORD}" == "" ]]; do
+        echo "Not yet PGPASSWORD"
+        sleep 20
+        get_postgresql_postgres_password
+      done
+
+      # test1 -- authenticated connection
+      if ! PGPASSWORD="${PGPASSWORD}" psql sslmode=require -U postgres -h "${IP}" -c '\dt' | grep -q 'No relations found'; then 
+        echo "Test1: secure connection -- failed"
+        $((FAILED+=1))
+      fi
+
+      if [[ "${FAILED}" != 0 ]]; then
+        echo
+        echo "Username: postgres"
+        echo "Password: ${PGPASSWORD}"
+        echo "IP address: ${IP}"
+        echo
+        exit "${FAILED}"
+      fi
+    expect:
+      exitCode:
+        equals: 0
+- name: db operations
+  bashTest:
+    script: |-
+      SERVICE="${APP_INSTANCE_NAME}-postgresql-svc"
+      IP=''
+      PGPASSWORD=''
+      FAILED=0
+
+      function wait_for_service_external_ip {
+        IP=$(kubectl get svc "${SERVICE}" \
+          --namespace "${NAMESPACE}" \
+          -ojsonpath="{.status.loadBalancer.ingress[0].ip}")
+        if [[ $? -ne 0 ]]; then
+          IP=""
+        fi
+      }
+
+      function get_postgresql_postgres_password {
+        PGPASSWORD=$(kubectl get secret "${APP_INSTANCE_NAME}-secret" \
+          --namespace "${NAMESPACE}" \
+          --output=jsonpath='{.data.password}' | base64 -d)
+        if [[ $? -ne 0 ]]; then
+          PGPASSWORD=""
+        fi
+      }
+
+      echo "Waiting for Service external ip address."
```

Comment:
Tests are executed sequentially, we do not need to have these waiters.

Reply:
Testing environment is not reliable.

Without waiters if someone removed first tests, it would require to change the second test.